### ISSUE TITLE
No internet connectivity using rootless Podman with pasta networking

### ISSUE THREAD

**Issue by skyblaster** [coreos/fedora-coreos-tracker#1790]
### Describe the bug

It appears that after my host updated from 40.20240709.3.1 to 40.20240728.3.0, rootless containers lost Internet connectivity. I am now running 40.20240808.3.0 and the problem persists.

First noticed after my changedetection.io container started failing with `Exception: net::ERR_NAME_NOT_RESOLVED` on Aug. 17th after my Zincati maintenance window.

### Reproduction steps

```
$ podman network create test
$ podman run -d --rm --network test --name netshoot docker.io/nicolaka/netshoot sleep 600
$ podman exec -it netshoot /bin/sh
~ # ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
From 10.89.3.1 icmp_seq=1 Destination Net Unreachable
From 10.89.3.1 icmp_seq=2 Destination Net Unreachable
From 10.89.3.1 icmp_seq=3 Destination Net Unreachable
~ # ping 10.89.3.1
PING 10.89.3.1 (10.89.3.1) 56(84) bytes of data.
64 bytes from 10.89.3.1: icmp_seq=1 ttl=64 time=0.106 ms
64 bytes from 10.89.3.1: icmp_seq=2 ttl=64 time=0.099 ms
```

### Expected behavior

```
~ # ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
64 bytes from 8.8.8.8: icmp_seq=1 ttl=254 time=7.93 ms
64 bytes from 8.8.8.8: icmp_seq=2 ttl=254 time=7.94 ms
64 bytes from 8.8.8.8: icmp_seq=3 ttl=254 time=8.37 ms
```

### Actual behavior

```
~ # ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
From 10.89.3.1 icmp_seq=1 Destination Net Unreachable
From 10.89.3.1 icmp_seq=2 Destination Net Unreachable
From 10.89.3.1 icmp_seq=3 Destination Net Unreachable
~ # ping 10.89.3.1
PING 10.89.3.1 (10.89.3.1) 56(84) bytes of data.
64 bytes from 10.89.3.1: icmp_seq=1 ttl=64 time=0.106 ms
64 bytes from 10.89.3.1: icmp_seq=2 ttl=64 time=0.099 ms
```

### System details

Bare Metal
```
● fedora:fedora/x86_64/coreos/stable
                  Version: 40.20240808.3.0 (2024-08-27T00:24:44Z)
                   Commit: ba0b39f81de5a42c61b93514edfeeeecafa206f17f6509b9616a70be0d150897
             GPGSignature: Valid signature by 115DF9AEF857853EE8445D0A0727707EA15B79CC
```

### Additional information

Running Podman 5.2.0 on a project Bluefin host, I'm able to ping out to the Internet following the reproduction steps.

Also confirmed working with Podman 5.2.1 on GCP:
```
● fedora:fedora/x86_64/coreos/next
                  Version: 40.20240825.1.0 (2024-08-27T00:24:46Z)
                   Commit: e0b348eb6b38315018c23107e5424156584044676946e0e7491a5c33bba22d40
             GPGSignature: Valid signature by 115DF9AEF857853EE8445D0A0727707EA15B79CC
```

**Comment by dustymabe** [comment]
TBH I'm not really sure if ICMP (ping) is supposed to work in this setup, but at least actually getting out to the internet (i.e. things like DNS, HTTP, etc) seem to be working fine for me:

```
Fedora CoreOS 40.20240808.3.0
Tracker: https://github.com/coreos/fedora-coreos-tracker
Discuss: https://discussion.fedoraproject.org/tag/coreos

Last login: Thu Sep  5 01:51:34 2024
[core@cosa-devsh ~]$ 
[core@cosa-devsh ~]$ 
[core@cosa-devsh ~]$ podman network create test
test
[core@cosa-devsh ~]$ podman run -d --rm --network test --name netshoot docker.io/nicolaka/netshoot sleep 600
Trying to pull docker.io/nicolaka/netshoot:latest...
Getting image source signatures
Copying blob b8cdfec6d24e done   | 
Copying blob ff793c57efef done   | 
Copying blob f72249ed6705 done   | 
Copying blob b6621d484422 done   | 
Copying blob 4abcf2066143 done   | 
Copying blob d21093198226 done   | 
Copying blob 452eb7889eb5 done   | 
Copying blob 4f4fb700ef54 done   | 
Copying blob 89065cf5c037 done   | 
Copying blob a4b421d4901a done   | 
Copying blob d5c3ad7ea15a done   | 
Copying blob ab073295bbd0 done   | 
Copying blob 737c1bf9f2ef done   | 
Copying blob 097ac21093f8 done   | 
Copying blob 59e353e0ee74 done   | 
Copying config 27b858cdcd done   | 
Writing manifest to image destination
71fc86bd0d33270da8be4d5a70f0b1de951ca587c2b2a9507f9d49429d7b646b
[core@cosa-devsh ~]$ podman exec -it netshoot /bin/sh
~ # ping 8.8.8.8
PING 8.8.8.8 (8.8.8.8) 56(84) bytes of data.
^C
--- 8.8.8.8 ping statistics ---
123 packets transmitted, 0 received, 100% packet loss, time 124912ms

~ # curl google.com
<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>301 Moved</TITLE></HEAD><BODY>
<H1>301 Moved</H1>
The document has moved
<A HREF="http://www.google.com/">here</A>.
</BODY></HTML>
~ # dig +short google.com
172.253.63.138
172.253.63.100
172.253.63.113
172.253.63.139
172.253.63.102
172.253.63.101
```

**Comment by skyblaster** [comment]
> TBH I'm not really sure if ICMP (ping) is supposed to work in this setup

ICMP works from my FCOS 40.20240825.1.0 install in GCP.

>but at least actually getting out to the internet (i.e. things like DNS, HTTP, etc) seem to be working fine for me:

Very interesting that it works for you. DNS resolution and curl both fail for me.

```
~ # curl google.com
curl: (6) Could not resolve host: google.com

~ # dig google.com
;; communications error to 10.89.3.1#53: timed out
;; communications error to 10.89.3.1#53: timed out
;; communications error to 10.89.3.1#53: timed out

~ # dig @8.8.8.8 google.com
;; communications error to 8.8.8.8#53: timed out
;; communications error to 8.8.8.8#53: timed out
;; communications error to 8.8.8.8#53: timed out

~ # curl 142.250.69.206
curl: (7) Failed to connect to 142.250.69.206 port 80 after 0 ms: Couldn't connect to server
```

>What's the actual error you are seeing?

The error is that after my last scheduled maintenance reboot, my rootless changedetection.io container could no longer reach any websites in it's list.

I suppose I could rebase to testing, but maybe there are other things you could suggest before doing so.

**Comment by skyblaster** [comment]
Podman info for reference.
```
$ podman info
host:
  arch: amd64
  buildahVersion: 1.36.0
  cgroupControllers:
  - cpu
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.10-1.fc40.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.10, commit: '
  cpuUtilization:
    idlePercent: 98.93
    systemPercent: 0.54
    userPercent: 0.53
  cpus: 4
  databaseBackend: boltdb
  distribution:
    distribution: fedora
    variant: coreos
    version: "40"
  eventLogger: journald
  freeLocks: 2045
  hostname: falcon
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 6.9.12-200.fc40.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 8705351680
  memTotal: 12347645952
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.12.1-1.fc40.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.12.1
    package: netavark-1.12.1-1.fc40.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.12.1
  ociRuntime:
    name: crun
    package: crun-1.15-1.fc40.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.15
      commit: e6eacaf4034e84185fd8780ac9262bbf57082278
      rundir: /run/user/1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt-0^20240726.g57a21d2-1.fc40.x86_64
    version: |
      pasta 0^20240726.g57a21d2-1.fc40.x86_64
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: false
    path: /run/user/1000/podman/podman.sock
  rootlessNetworkCmd: pasta
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.2-2.fc40.x86_64
    version: |-
      slirp4netns version 1.2.2
      commit: 0ee2d87523e906518d34a6b423271e4826f71faf
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.5
  swapFree: 0
  swapTotal: 0
  uptime: 103h 6m 31.00s (Approximately 4.29 days)
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
store:
  configFile: /var/home/core/.config/containers/storage.conf
  containerStore:
    number: 2
    paused: 0
    running: 2
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/home/core/.local/share/containers/storage
  graphRootAllocated: 511561764864
  graphRootUsed: 22538981376
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Supports shifting: "false"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 6
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /var/home/core/.local/share/containers/storage/volumes
version:
  APIVersion: 5.1.2
  Built: 1720569600
  BuiltTime: Wed Jul 10 00:00:00 2024
  GitCommit: ""
  GoVersion: go1.22.5
  Os: linux
  OsArch: linux/amd64
  Version: 5.1.2

```

**Comment by skyblaster** [comment]
I opted for the rebase to testing and things are back to normal.

**Comment by dustymabe** [comment]
> I opted for the rebase to testing and things are back to normal. ICMP, DNS, and HTTP are all working.

Well that's good news. I guess a regression that was then fixed when moving from podman 5.1.2 to 5.2.1.
